internal/k8shandler/collection.go

```diff
@@ -112,13 +112,21 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCollection() (err err
 	var httpInputs []string
 	for _, input := range clusterRequest.Forwarder.Spec.Inputs {
 		if input.Receiver != nil && input.Receiver.HTTP != nil {
-			httpInputs = append(httpInputs, input.Name)
+			serviceName := fmt.Sprintf("%s-%s-%s", input.Name, clusterRequest.Forwarder.Namespace, clusterRequest.Forwarder.Name)
+			httpInputs = append(httpInputs, serviceName)
 		}
 	}
 
-	if err := factory.ReconcileDaemonset(clusterRequest.EventRecorder, clusterRequest.Client, clusterRequest.Forwarder.Namespace, clusterRequest.ResourceOwner, httpInputs); err != nil {
-		log.Error(err, "collector.ReconcileDaemonset")
-		return err
+	if clusterRequest.isDaemonset {
```

Comment:
what if the someone edits the CLF to add more inputs?  do we lose ref to the previous deployment which may lead us to having multiple collectors running?

Reply:
When any input is added to a pipeline that isn't an HTTP receiver, it will reconcile into a daemonset and remove the deployment.

Reply:
Please make sure to add this as a testcase to the issue we need to check